lib/miou_unix.ml

```diff
@@ -367,25 +366,25 @@ let select uid (interrupt, p) ~block cancelled_syscalls =
       if interrupted interrupt rds then begin
         Logs.debug (fun m ->
             m "[%a] interrupted by Miou" Miou.Domain.Uid.pp uid);
-        intr interrupt p
+        intr interrupt
       end;
       let signals = collect domain [] in
       let signals = transmit_fds signals domain.revert domain.readers rds in
       let signals = transmit_fds signals domain.revert domain.writers wrs in
       signals
 
-let rec interrupt oc interrupted () =
-  if Atomic.get interrupted = 0 then
-    if Unix.single_write oc signal 0 1 = 0 then interrupt oc interrupted ()
-    else ignore (Atomic.fetch_and_add interrupted 1)
+let interrupt oc () =
+  match Unix.single_write oc signal 0 1 with
+  | _ -> ()
```

Comment:
I'd test that I've actually written one byte, more for pedantry, as it's actually impossible due to the underlying implementation, but the basically writes on stream can/could always return shortcounts.